File: DS_progs/binary_search.cpp
#include <iostream>
using namespace std;

int binary_search(int A[], int x)
{
    int l = 0, h = sizeof(A) - 1;
    int mid = (l + h2);
    while (l <= h)
    {
        if (x == A[mid])
            return mid;
        else if (x > A[mid])
            l = mid + 1;
        else
            h = mid – 1;
    }
    return -1;
}

int main(void)
{
    int arr[] = {32, 21, 65, 35, 69, 75, 9, 84};
    int x = 32;
    int index = binary_search(arr, x);
    if (index == -1)
        cout << x << " is not present in the array";
    else
        cout << x << " is present at index " << index << " in the array";
    return 0;
}

File: Javascript_DSALGO/Search Algorithms/binarySearch.js
function binarySearch(arr, val){
  // add whatever parameters you deem necessary - good luck!
  let l = 0, r = arr.length - 1, mid = Math.floor((r + l) / 2)
  
  while(arr[mid] !== val && l <= r){
      if(arr[mid] < val) l = mid + 1
      else r = mid - 1;
        
      mid = Math.floor((r + l) / 2)
  }
  return arr[mid] == val ? mid : -1
}

binarySearch([1, 3, 5, 8, 9, 15], 8)